build-tools/packages/build-tools/src/type-test-generator/typeTestUtils.ts

```diff
@@ -0,0 +1,281 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { readJsonSync } from "fs-extra";
+import path from "node:path";
+import { ExtractorConfig } from "@microsoft/api-extractor";
+import { existsSync } from "node:fs";
+import { Project, SourceFile } from "ts-morph";
+import { BrokenCompatTypes } from "../common/fluidRepo";
+import { buildTestCase, TestCaseTypeData } from "../typeValidator/testGeneration";
+import { getFullTypeName, getNodeTypeData, TypeData } from "../typeValidator/typeData";
+import { PackageJson } from "../common/npmPackage";
+
+// Information about the previous package from the package.json is not needed,
+// but error if it's missing since it's nice to separate errors for the dep missing here vs not installed.
+// This ensures that a critical dependency (the previous package version) is correctly declared in the project's package.json.
+export function ensureDevDependencyExists(
+	packageObject: PackageJson,
+	dependencyName: string,
+): void {
+	const dependencyVersion = packageObject?.devDependencies?.[dependencyName];
+	if (typeof dependencyVersion !== "string") {
+		throw new Error(`Did not find devDependency ${dependencyName} in package.json`);
+	}
+}
+
+/**
+ * Fetches the path of the previous package.json or throws an error if not found.
+ * @param previousBasePath - A string representing the path to the root of a package
+ * @returns
+ */
+export function getPreviousPackageJsonPath(previousBasePath: string): string {
+	const previousPackageJsonPath = path.join(previousBasePath, "package.json");
+	if (!existsSync(previousPackageJsonPath)) {
+		throw new Error(`${previousPackageJsonPath} not found.`);
+	}
+	return previousPackageJsonPath;
+}
+
+/**
+ * Attempts to retrieve  a specified type of rollup file path for type definitions from the API Extractor configuration.
+ * @param {string} rollupType - The type of rollup file path to retrieve (ex: "alpha", "beta", "public").
```

Comment:
Since this is TypeScript code, the type annotations aren't needed for the param comments, etc.

```suggestion
 * @param rollupType - The type of rollup file path to retrieve (ex: "alpha", "beta", "public").
```